### PR TITLE
Add ability to filter by other columns' values besides "Name" in Jobs List

### LINKED ISSUE #528: Add in "filter by" menu to all (appropriate) cells in job list table
Currently, the user can only click on a job list cell and filter by the name:value of that cell in the "Name" column.  This functionality should be configurable so it can be added to other columns, as well.

Functionality shown in this mockup:
screen shot 2018-12-03 at 9 59 46 am

### PR DESCRIPTION
**Pull request by rsasch**
- Make handling of `Name` column more general (both UI and if using a config file, it needs to be explicitly included)
- If a `DisplayField` is of type `text` and is not `editable`, add the ability to filter by `name:value`

screen shot 2018-12-03 at 11 51 17 am

These changes are also in anticipation of making the ordering of columns in the jobs list configurable.

Closes #528

### CODE REVIEW

File: ui/src/app/job-list/table/table.component.html [hunk 1]
```diff
@@ -64,30 +52,30 @@
...
-                    <a (click)="filterOnFieldValue('name', j.name)" clrTooltipTrigger>
-                      <clr-icon shape="filter-2" size="24"></clr-icon>
-                      <span class="job-title-filter">Filter by name:{{ j.name }}</span>
-                      <clr-tooltip-content clrPosition="top-right" clrSize="xs" *clrIfOpen>
-                        <span>Filter by name:{{ j.name }}</span>
-                      </clr-tooltip-content>
-                    </a>
-                  </clr-tooltip>
-              </mat-list-item>
+              <mat-list-item *ngIf="j[displayedColumns[1]] && canFilterBy(displayedColumns[1])">
```
Comment (bfcrampton) [inline ui/src/app/job-list/table/table.component.html]: nit: could we extract out `displayedColumns[1]` in `setUpFieldsAndColumns` to make it clearer what it is and shorten the html.

File: ui/src/app/job-list/table/table.component.html [hunk 2]
```diff
@@ -64,30 +52,30 @@
...
   </ng-container>
 
-  <!-- Implementation-specific additional columns -->
+  <!-- Implementation-specific columns -->
   <ng-container *ngFor="let df of displayFields">
     <ng-container matColumnDef="{{ df.field }}">
       <mat-header-cell *matHeaderCellDef>{{ df.display }}</mat-header-cell>
       <mat-cell class="additional-column" *matCellDef="let j">
-        <div class="display-field" *ngIf=canEdit(df)>
+        <div class="display-field" *ngIf="canEdit(df) || (canFilterBy(df.field) && (getFieldValue(j, df) != ''))">
```
Comment (bfcrampton) [inline ui/src/app/job-list/table/table.component.html]: tiny nit: I think the extra parens around `(getFieldValue(j, df) != '')` are unecessary:
```
... || (canFilterBy(df.field) && getFieldValue(j, df) != '')
```

File: ui/src/app/job-list/table/table.component.html [hunk 3]
```diff
@@ -123,7 +124,12 @@
             </mat-form-field>
           </div>
         </div>
-        <div class="display-field" *ngIf="!isStatusField(df) && !canEdit(df)">{{ getFieldValue(j, df) }}</div>
+        <div class="display-field" *ngIf="isFirstColumn(df)">
+          <a *ngIf="isFirstColumn(df)" class="field-value job-details-button mat-body-1"
+             [routerLink]="[j.id]"
+             [queryParams]="getQueryParams()">{{ getFieldValue(j, df) }}</a>
+        </div>
+        <div class="display-field" *ngIf="!isStatusField(df) && !canEdit(df) && !isFirstColumn(df) && !canFilterBy(df.field)">{{ getFieldValue(j, df) }}</div>
```
Comment (bfcrampton) [inline ui/src/app/job-list/table/table.component.html]: nit: This is a pretty long block of code for an `*ngIf`, could you create a wrapper function which calls all these?

**Comment by rsasch** [comment]
@rexwangcc Yup, it will – if only to explicitly add the `Name` display column.

File: ui/src/app/job-list/table/table.component.html (hunk 1 above)
Comment (rsasch) [reply ui/src/app/job-list/table/table.component.html]: sure – done

File: ui/src/app/job-list/table/table.component.html (hunk 2 above)
Comment (rsasch) [reply ui/src/app/job-list/table/table.component.html]: good point – simplified

File: ui/src/app/job-list/table/table.component.html (hunk 3 above)
Comment (rsasch) [reply ui/src/app/job-list/table/table.component.html]: done

**Pull request merged**